truss/remote/baseten/core.py

```diff
@@ -271,7 +272,7 @@ def create_truss_service(
         return model_version_json["id"], model_version_json["version_id"]
 
     if model_id is None:
-        if environment:
+        if environment != PRODUCTION_ENVIRONMENT_NAME:
```

Comment:
is it possible for `environment` to be `None` here? we should handle that as well, maybe something like 
```suggestion
        if environment and environment != PRODUCTION_ENVIRONMENT_NAME:
```

Reply:
yeah good catch. I saw that as I was writing tests as well. I think we should be good now.